script/tool/lib/src/format_command.dart

```diff
@@ -212,4 +241,65 @@ class FormatCommand extends PluginCommand {
 
     return javaFormatterPath;
   }
+
+  /// Returns true if [command] can be run successfully.
+  Future<bool> _hasDependency(String command) async {
+    try {
+      final io.ProcessResult result =
+          await processRunner.run(command, <String>['--version']);
+      if (result.exitCode != 0) {
+        return false;
+      }
+    } on io.ProcessException {
+      // Thrown when the binary is missing entirely.
+      return false;
+    }
+    return true;
+  }
+
+  /// Runs [command] on [arguments] on all of the files in [files], batched as
+  /// necessary to avoid OS command-line length limits.
+  ///
+  /// Returns the exit code of the first failure, which stops the run, or 0
+  /// on success.
+  Future<int> _runBatched(
+    String command,
+    List<String> arguments, {
+    required Iterable<String> files,
+  }) async {
+    final int commandLineMax =
+        platform.isWindows ? windowsCommandLineMax : nonWindowsCommandLineMax;
+
+    // Compute the max length of the file argument portion of a batch.
+    // Add one to each argument's length for the space before it.
+    final int argumentTotalLength =
+        arguments.fold(0, (int sum, String arg) => sum + arg.length + 1);
+    final int batchMaxTotalLength =
+        commandLineMax - command.length - argumentTotalLength;
+
+    // Divide the list into batches that don't exceed the length.
```

Comment:
Maybe split this to its own method, just to make _runBatched super straight-forward:

final batches = _createBatches(params);
for (batch in batches) ...

Reply:
Good idea, done.